### PR TITLE
Fix issues raised by golosio

### PR DESCRIPTION
**Pull request by AlexVanMeegen**
Various fixes for the issues raised by @golosio in #21 and #22 and some additional minor fixes:
* add `pytest` to the requirements
* fix the `SIM_LABELS` access in `figures/Schmidt2018_dyn/Snakefile` (see #22)
* fix division by zero errors in `analysis_helpers.py` (see #21)
* fix default value of `K_stable` (see #21)

@jarsi, could you have a quick look?

### CODE REVIEW

File: multiarea_model/analysis_helpers.py [hunk 1]
```diff
@@ -668,8 +668,11 @@ def synchrony(data_array, num_neur, t_min, t_max, resolution=1.0):
     spike_count_histogramm = pop_rate_time_series(
         data_array, num_neur, t_min, t_max, resolution=resolution)
     mean = np.mean(spike_count_histogramm)
-    variance = np.var(spike_count_histogramm)
-    synchrony = variance / mean
+    std_dev = np.std(spike_count_histogramm)
+    if std_dev + mean != std_dev:  # check for vanishing mean
```
Comment (jarsi) [inline multiarea_model/analysis_helpers.py:672]: Maybe I am a bit pedantic here, but addition is limited by floating point precision. E.g.
```
In [1]: 0.1 + 0.2                                                               
Out[1]: 0.30000000000000004

In [2]: 0.1 + 0.2 == 0.3                                                        
Out[2]: False

```
I don't know if these floating point errors can also happen when you add 0. What about?
```
try:
    synchrony = std_dev / mean
except ZeroDivisionError:
    synchrony = np.inf
```

**Review by jarsi — COMMENTED** [review]
Thanks for tackling these issues 🔥

It all looks good, one point might need discussion.

File: multiarea_model/analysis_helpers.py (hunk 1 above)
Comment (golosio) [reply multiarea_model/analysis_helpers.py:672]: Thank you @AlexVanMeegen and @jarsi,
> I don't know if these floating point errors can also happen when you add 0
Just for information, actually that kind of check is motivated exactly by the fact that addition is limited by floating point precision. As I said, it's a standard trick, you can find it e.g. in "Numerical recipes in C. The Art of Scientific Computing". a + b should always be == a if b is 0 or smaller than the floating point precision of a. That said, for sure your solution is much more elegant!

Comment (AlexVanMeegen) [reply multiarea_model/analysis_helpers.py]: Also think your solution, @jarsi, is cleaner. Thanks for being pendantic :nerd_face:

**Comment by AlexVanMeegen** [comment]
Addressed your concern, @jarsi. Please merge if you deem it appropriate.

**Comment by jarsi** [comment]
Thanks for the work, merged.

**Comment by AlexVanMeegen** [comment]
Thanks for reviewing!

**Pull request merged**